snake/snake.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/python3.11
+
+class SnakeGame:
+
+    def __init__(self, width: int, height: int, food: list[list[int]]):
+        self.snake_loc = [(0,0)]
+        self.width = width
+        self.height = height
+        self.food = food
+
+
+    def move(self, direction: str) -> int:
+        score = len(self.snake_loc) - 1
+        if direction == 'R':
+            new_head_loc = (self.snake_loc[0][0], self.snake_loc[0][1] + 1)
+        elif direction == 'D':
+            new_head_loc = (self.snake_loc[0][0] + 1, self.snake_loc[0][1])
+        elif direction == 'L':
+            new_head_loc = (self.snake_loc[0][0], self.snake_loc[0][1] - 1)
+        elif direction == 'U':
+            new_head_loc = (self.snake_loc[0][0] - 1, self.snake_loc[0][1])
+        else:
+            raise 'Got invalid direction'
+
+        if new_head_loc in self.snake_loc[:-1] or \
+           new_head_loc[0] < 0 or new_head_loc[0] >= self.height or \
+           new_head_loc[1] < 0 or new_head_loc[1] >= self.width:
+            return -1
+
+        if score < len(self.food) and new_head_loc[0] == self.food[score][0] and new_head_loc[1] == self.food[score][1]:
+            # eating the food
+            self.snake_loc.insert(0, new_head_loc)
+            score += 1
+        else:
+            # regular move no food
+            self.snake_loc.insert(0, new_head_loc)
+            self.snake_loc.pop(score + 1)
+        
+        return score
+
+def main() -> None:
+    ''' Main function '''
+    game = SnakeGame(2,2,[[0,1]])
```

Comment:
Not sure I understood in the code where you handle the case where the snake collides with itself, a wrong move in a sense means that game is over (snake hits the wall, or snake eats itself)

Reply:
new_head_loc in self.snake_loc[:-1] in line 13